evap/student/views.py

```diff
@@ -132,11 +132,10 @@ def get_valid_form_groups_or_render_vote_page(request, evaluation, preview, for_
             for questionnaire in questionnaires
         ]
 
-    if all(all(form.is_valid() for form in form_group) for form_group in form_groups.values()):
-        assert not preview
+    if not preview and all(all(form.is_valid() for form in form_group) for form_group in form_groups.values()):
```

Comment:
It took me some time to understand the `not preview` condition here - I find it confusing that the `preview` parameter influences the return type. Intuitively, the parameter should change the content of the returned value, not its shape. This whole `get_valid_form_groups_or_render_vote_page` thing is a bit weird anyways - there is only one place that actually uses the `form_group` result, all the others use `[1]` on the returned tuple. I suppose, we could refactor this to one function that builds the formgroups and another one that returns the rendered page, given the formgroups. We could make the form group argument optional (and default to the result of calling the former (pun intended)) to keep caller code short